mis_builder/models/mis_report_instance.py

```diff
@@ -669,10 +669,19 @@ def _compute_matrix(self):
         is guaranteed to be the id of the mis.report.instance.period.
         """
         self.ensure_one()
-        aep = self.report_id._prepare_aep(
-            self.query_company_ids, self.currency_id)
         kpi_matrix = self.report_id.prepare_kpi_matrix()
         for period in self.period_ids:
+
+            account_model = None
+            if period.source == 'actuals_alt' and period.source_aml_model_id:
+                account_field = period.source_aml_model_id.field_id.filtered(
+                    lambda r: r.name == 'account_id')
+                account_model = account_field.relation
+
+            aep = self.report_id._prepare_aep(self.query_company_ids,
+                                              self.currency_id,
+                                              account_model=account_model)
```

Comment:
Looking at this in more details, I think it does not make sense to have a different account_model for different columns, because of account details expansion which would become meaningless if different account objects are used in different columns.

Reply:
IOW, this may work, but the account model must be configured globally at the level of mis.report, and passed to the AEP constructor, as well as to the KPI matrix constructor.

Reply:
Or, better, define the AML model used for Actuals at the mis.report.instance level, and make sure Actual (alternative) models used in columns have the same account model in the account_id m2o.

Reply:
Another way to say it is that one way or another, the same AEP must be used for all columns and the account model must therefore be known before the `for period` loop.